devito/ir/cgen/printer.py

```diff
@@ -318,15 +406,21 @@ def _print_Fallback(self, expr):
 
 # Lifted from SymPy so that we go through our own `_print_math_func`
 for k in ('exp log sin cos tan ceiling floor').split():
-    setattr(CodePrinter, '_print_%s' % k, CodePrinter._print_math_func)
+    setattr(BasePrinter, f'_print_{k}', BasePrinter._print_math_func)
 
 
 # Always parenthesize IntDiv and InlineIf within expressions
 PRECEDENCE_VALUES['IntDiv'] = 1
 PRECEDENCE_VALUES['InlineIf'] = 1
 
 
-def ccode(expr, **settings):
+# Sympy 1.11 has introduced a bug in `_print_Add`, so we enforce here
+# to always use the correct one from our printer
+if Version(sympy.__version__) >= Version("1.11"):
+    setattr(sympy.printing.str.StrPrinter, '_print_Add', BasePrinter._print_Add)
+
+
+def ccode(expr, printer=None, **settings):
```

Comment:
if we move this function inside `ir/iet` probably we can avoid the internal import